File: src/main/java/eu/mikroskeem/moarsms/Platform.java
package eu.mikroskeem.moarsms;

import java.util.Map;

public interface Platform {
    /**
     * Shut down service handler
     * Currently used only in CLI version
     */
    void shutdown();

    /**
     * Get defined services and their secrets
     *
     * @return Map of service and it's secret
     */
    Map<String,String> getServiceSecrets();

    /**
     * Invoke service
     *
     * @param serviceId Service ID
     * @param message message in SMS (after keyword)
     * @return Reply to send to buyer
     */
    String invokeService(String serviceId, String message);

    /**
     * Test switch
     *
     * @return Whether test messages are allowed or not
     */
    boolean allowTest();

    /**
     * Get response messages
     *
     * @return Response message for service
     */
    String getMessage(String path);
}

File: src/main/kotlin/eu/mikroskeem/moarsms/Platform.kt
/*
 * Copyright (c) 2017 Mark Vainomaa
 *
 * This source code is proprietary software and must not be distributed and/or copied without the express permission of Mark Vainomaa
 */

package eu.mikroskeem.moarsms

import java.util.logging.Logger

/**
 * @author Mark Vainomaa
 */
internal interface Platform {
    val fortumoUtils: FortumoUtils
    val serviceSecrets: Map<String, String>
    val allowTest: Boolean
    val logger: Logger
    val defaultResponse: String
    val nettyLoggingLevel: String
    fun getMessage(path: String): String
    fun invokeService(serviceId: String, message: String): String
}